### PR TITLE
Refact test dir

### LINKED ISSUE #3995: Refactor 0_stateless
**The problem**

https://github.com/datafuselabs/databend/tree/main/tests/suites/0_stateless
has more and more tests, it's hard to category a new test case to the right group.

**Goals**
1. Add sub directory to `0_stateless` for group
`00_dummy` -- dummy
`01_system` -- system tables test
`02_function` -- functions test
`03_dml` -- dml test
`04_ddl` -- ddl test
`05_show` -- show test
`06_use` -- use test
`07_meta` -- meta related test
`08_fuse` -- fuse engine test

### PR DESCRIPTION
**Pull request by wangzhen11aaa**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Refact the stateless test structure

## Changelog
- Improvement

## Related Issues

Fixes #3995

## Test Plan

Unit Tests
Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
Cool.
Would like to know how the PR  format(PULL_REQUEST_TEMPLATE) was modified?

**Comment by wangzhen11aaa** [comment]
> 

ok

File: tests/suites/0_stateless/README.md [hunk 1]
```diff
@@ -9,16 +9,16 @@ xx_yyyy_[test_name]
...
+* 03_dml -- for Select tests, [example](03_0000_select_aliases.sql)
+* 04_ddl -- for Explain tests, [example](04_0000_explain.sql)
+* 05_show -- for DDL tests, [example](05_0000_ddl_create_tables.sql)
+* 06_use -- for Show tests, [example](06_0000_show_queries.sql)
+* 07_meta -- for Use tests, [example](07_0000_use_database.sql)
+* 08_fuse -- for Optimizer tests, [example](08_0000_optimizer.sql)
+* 09+_others -- for Other tests, [example](09_0000_remote_create_table.sql, 10_0000_describe_table.sql), etc
 * 10 -- for Describe tests, [example](10_0000_describe_table.sql)
 * 11 -- for Show tables, [example](11_0000_show_tables.sql)
 * 12 -- for Insert into, [example](12_0000_insert_into_select.sql)
```
Comment (PsiACE) [inline tests/suites/0_stateless/README.md:24]: Need to explain 09 . Also, if we merge these into one folder, we need to describe it by directory structure.

```suggestion
* 09+_others -- for Other tests, [example](09_0000_remote_create_table.sql, 10_0000_describe_table.sql), etc
  * 09 -- for Remote engine tests, [example](09_0000_remote_create_table.sql)
  * 10 -- for Describe tests, [example](10_0000_describe_table.sql)
  * 11 -- for Show tables, [example](11_0000_show_tables.sql)
  * 12 -- for Insert into, [example](12_0000_insert_into_select.sql)
```

**Comment by wangzhen11aaa** [comment]
Need help.
 
ERROR: responses 0.17.0 has requirement urllib3>=1.25.10, but you'll have urllib3 1.25.8 which is incompatible.
Installing collected packages: python-dateutil, jmespath, botocore, s3transfer, boto3, pytz, xmltodict, pycparser, cffi, cryptography, werkzeug, responses, jsondiff, wrapt, future, aws-xray-sdk, pbr, sarif-om, junit-xml, aws-sam-translator, jsonpickle, jschema-to-python, networkx, cfn-lint, graphql-core, websocket-client, docker, rsa, ecdsa, python-jose, sshpubkeys, moto, yapf, shfmt-py, mysql-connector
ERROR: aws-sam-translator 1.42.0 has requirement six~=1.15, but you'll have six 1.14.0 which is incompatible.
ERROR: cfn-lint 0.57.0 has requirement pyyaml>5.4, but you'll have pyyaml 5.3.1 which is incompatible.

How does the library incompatible problem happen? thanks advance.

**Comment by wangzhen11aaa** [comment]
> Need help.
> 
> ERROR: responses 0.17.0 has requirement urllib3>=1.25.10, but you'll have urllib3 1.25.8 which is incompatible. Installing collected packages: python-dateutil, jmespath, botocore, s3transfer, boto3, pytz, xmltodict, pycparser, cffi, cryptography, werkzeug, responses, jsondiff, wrapt, future, aws-xray-sdk, pbr, sarif-om, junit-xml, aws-sam-translator, jsonpickle, jschema-to-python, networkx, cfn-lint, graphql-core, websocket-client, docker, rsa, ecdsa, python-jose, sshpubkeys, moto, yapf, shfmt-py, mysql-connector ERROR: aws-sam-translator 1.42.0 has requirement six~=1.15, but you'll have six 1.14.0 which is incompatible. ERROR: cfn-lint 0.57.0 has requirement pyyaml>5.4, but you'll have pyyaml 5.3.1 which is incompatible.
> 
> How does the library incompatible problem happen? thanks advance.

I think we will change the requirements use by python now.

**Comment by PsiACE** [comment]
> How does the library incompatible problem happen? thanks advance.

See .github/action/test_stateless_*

```yaml
    - name: Install python dependences
      shell: bash
      run: |
        pip3 install --user boto3 "moto[all]" yapf shfmt-py mysql-connector
```

We actually use poetry to provide the lock file. Perhaps poetry should be used in the ci as well.

**Comment by wangzhen11aaa** [comment]
> > How does the library incompatible problem happen? thanks advance.
> 
> See .github/action/test_stateless_*
> 
> ```yaml
>     - name: Install python dependences
>       shell: bash
>       run: |
>         pip3 install --user boto3 "moto[all]" yapf shfmt-py mysql-connector
> ```
> 
> We actually use poetry to provide the lock file. Perhaps poetry should be used in the ci as well.

ok

**Review by PsiACE — APPROVED** [review]
lgtm

**Comment by BohuTANG** [comment]
@mergifyio update

**Comment by BohuTANG** [comment]
Cool, but some directory tests is not in right places, for example: show directory is ddl tests.

**Comment by BohuTANG** [comment]
Like `05_show` -- Should be for `show` statements(not `05_show -- for DDL tests`), every direcotry is the stateless test prefix.
BTW, after merged with main branch by mergeify bot, there is a `06_0003_show_settings.sql` test alone, we should move it to the `05_show` :
image

File: tests/suites/0_stateless/00_dummpy/00_0000_dummy_select_py.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from moto import mock_s3
```
Comment (BohuTANG) [inline tests/suites/0_stateless/00_dummpy/00_0000_dummy_select_py.py:8]: `00_dummpy`-> `00_dummy`

**Comment by wangzhen11aaa** [comment]
> Like `05_show` -- Should be for `show` statements(not `05_show -- for DDL tests`), every direcotry is the stateless test prefix. BTW, after merged with main branch by mergeify bot, there is a `06_0003_show_settings.sql` test alone, we should move it to the `05_show` : image

Ok.

**Comment by BohuTANG** [comment]
/LGTM

Thank you @wangzhen11aaa , 春节快乐！

**Comment by wangzhen11aaa** [comment]
> /LGTM
> 
> Thank you @wangzhen11aaa , 春节快乐！

新年快乐。
BTW, the README.md in the stateless directory is not the Markdown format. Let me fix it. Good start  with good end.

**Pull request merged**